### PR TITLE
Declare more non-static functions in header files

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

File: include/libc.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#ifndef GUARD_LIBC_H
+#define GUARD_LIBC_H
+
+void *memcpy(void *dest, const void *src, size_t n);
+void *memset(void *s, int c, size_t n);
+int strcmp(const char *s1, const char *s2);
+
+#endif // GUARD_LIBC_H
```
Comment (YamaArashi) [inline include/libc.h]: Please remove this header. This is covered by the standard C headers.

**Comment by marijnvdwerf** [comment]
```bash
make clean; make ruby; make compare;
...
pokeruby.gba: OK
```

**Comment by camthesaxman** [comment]
I think it would be a good idea to declare functions that take no parameters with void in the parentheses. This way, the compiler is able to check that no parameters were passed.

**Comment by YamaArashi** [comment]
Yeah. With the exception of GetMonData and GetBoxMonData, use `u8 sub_803DAA0(void);` instead of `u8 `sub_803DAA0();`, for example.

**Comment by camthesaxman** [comment]
Const correctness would be nice, too. If a function takes a pointer and doesn't modify the object pointed to, then it should be declared const.

**Comment by marijnvdwerf** [comment]
I need to include either `global.h` or `gba/types.h` in the headers for them to stop giving errors.

**Comment by camthesaxman** [comment]
If you include global.h at the very beginning of all the .c files, then all of the headers included after global.h will have those type definitions.

**Comment by marijnvdwerf** [comment]
@camthesaxman @YamaArashi Better?

**Pull request merged**